### PR TITLE
Adds listThreads and getUsers within the server proxy interface and controllers

### PR DESCRIPTION
**Pull request by FFDeveloper**

### CODE REVIEW

File: src/main/kotlin/com/safehill/kclient/network/ServerProxy.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                        publicKeyData = serverUser.publicKeyData,
+                        publicSignatureData = serverUser.publicSignatureData
+                    )
+                } catch (exception: Exception) {
+                    println("failed to create server user in local server: ${exception.message}")
+                }
+
+            }
+            if (serverUserChunk.isNotEmpty() && i < serverUserChunk.size - 1) {
+                delay(10) // sleep 10ms
```
Comment (gennaro-safehill) [inline src/main/kotlin/com/safehill/kclient/network/ServerProxy.kt]: Why are we sleeping 10ms here? If this is a copy of the Swift code, know that it was just an artifact of testing scaling performance, which should be removed (and has been partially removed recently) in the Swift codebase

File: src/main/kotlin/com/safehill/kclient/network/ServerProxy.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            if (localUsers.size == userIdentifiersToFetch.size) {
+                localUsers
+            } else {
+                throw exception
+            }
+        }
+    }
+
+    private suspend fun updateLocalUserDB(serverUsers: List<SHServerUser>): List<SHServerUser> {
+        serverUsers.chunked(5).forEachIndexed { i, serverUserChunk ->
```
Comment (gennaro-safehill) [inline src/main/kotlin/com/safehill/kclient/network/ServerProxy.kt]: I would try to remove this chunking logic

**Review by gennaro-safehill — COMMENTED** [review]
LGTM overall, just one comment

**Review by gennaro-safehill — APPROVED** [review]

File: src/main/kotlin/com/safehill/kclient/network/ServerProxy.kt (hunk 1 above)
Comment (FFDeveloper) [reply src/main/kotlin/com/safehill/kclient/network/ServerProxy.kt]: Changed

File: src/main/kotlin/com/safehill/kclient/network/ServerProxy.kt (hunk 2 above)
Comment (FFDeveloper) [reply src/main/kotlin/com/safehill/kclient/network/ServerProxy.kt]: Changed

**Pull request merged**